lib/p2p/Peer.ts

```diff
@@ -932,6 +934,12 @@ class Peer extends EventEmitter {
   }
 
   private handleDisconnecting = (packet: packets.DisconnectingPacket): void  => {
+    if (packet.body && packet.body.reason === DisconnectionReason.Banned) {
+      this.recvDisconnectionReason = packet.body.reason;
+      // if we are banned by the peer change `bannedBy` in DB.
+      this.emit('bannedBy', this.expectedNodePubKey!);
```

Comment:
We may have established the pub key for the peer (such that we're no longer merely "expecting" it) so this should be: `this._nodePubKey || this.expectedNodePubKey` - I'll update the code snippet above with this.